src/utilities/parse-pulp-id.ts

```diff
@@ -10,3 +12,195 @@ export function parsePulpIDFromURL(url: string): string {
 
   return null;
 }
+
+export function parsePulpIDFromPRN(prn: string): string {
+  // PRNs are of the form prn:app_label.model_label:pulp_id
+  const sections = prn.split(':');
+  if (sections.length == 3 && sections[2].match(UUIDRegEx)) {
+    return sections[2];
+  }
+
+  return null;
+}
+
+interface PulpResource {
+  id: string;
+  plugin: string;
+  model: string;
+  resource_label?: string;
+}
+
+export function parsePulpResource(value: string): PulpResource {
+  // value can be a url or a prn
+  let id = '',
+    plugin = '',
+    model = '',
+    resource_label;
+  if (value.startsWith(API_BASE_PATH)) {
+    const url = value.replace(API_BASE_PATH, '');
+    id = parsePulpIDFromPRN(url);
+    [plugin, model] = translateURLToModelLabel(url).split('.');
```

Comment:
@gerrod3 Calling ` translateURLToModelLabel(url)` on `domains/01907da2-e881-7c02-98f1-2118d9914b14/` or
`remotes/ansible/collection/0191eaf3-eaf2-7031-b02d-29ea4c67d5c3/` returns `undefined` and `split('.')` throws an error so detail page is redirected to 404. 


I'd add a check before this line
```diff
--- a/src/utilities/parse-pulp-id.ts
+++ b/src/utilities/parse-pulp-id.ts
@@ -39,6 +39,9 @@ export function parsePulpResource(value: string): PulpResource {
   if (value.startsWith(API_BASE_PATH)) {
     const url = value.replace(API_BASE_PATH, '');
     id = parsePulpIDFromPRN(url);
+    if(!translateURLToModelLabel(url)) {
+      return null;
+    };
     [plugin, model] = translateURLToModelLabel(url).split('.');
   } else if (value.startsWith('prn:')) {
     // prn is of the form `prn:app_label.model_label:pulp_id`
```


WDYT?

Otherwise looks good :)